### ISSUE TITLE
Filters broken in Python2.6.6

### ISSUE THREAD

**Issue by verbitan** [gcovr/gcovr#174]
I run the following command in gcovr3.3 without issue, however if I run the same command on the latest version I get no output.

> gcovr -r /path/to/coverage --branches --sort-percentage --gcov-exclude=".*Pattern.*" --html --html-details -o Coverage.html

I've traced this to commit 90589892b9f32d811bedca3008d1e8960389faf2. I added the following debugging statement which showed the error.

```
    sys.stdout.write(re.compile(options.gcov_exclude[i]).pattern)
    sys.stdout.write(re.compile(os.path.realpath(options.gcov_exclude[i])).pattern)
```

> /path/to/file/.*Pattern.*
> Traceback (most recent call last):
>   File "/usr/local/bin/gcovr", line 2286, in <module>
>     sys.stdout.write(re.compile(os.path.realpath(options.gcov_exclude[i])).pattern)
>   File "/usr/lib64/python2.6/posixpath.py", line 357, in realpath
>     if isabs(filename):
>   File "/usr/lib64/python2.6/posixpath.py", line 52, in isabs
>     return s.startswith('/')
> AttributeError: startswith

I haven't yet been able to try different versions of Python, although I couldn't see a minimum version specified. The closest I found was doc/guide.txt saying that gcovr has been tested with 2.7, 3.4 and 3.5.

Would it be possible to either fix the above issue, or check that the minimum supported version of Python is being used when executing gcovr?

**Comment by latk** [comment]
Could you clarify what precisely the problem is? 

 - Is the pattern not matching as expected, thus leading to the wrong files in the coverage report?

    If so, what directory structure do you have, which files do you want to filter, and which actual pattern are you using?

 - Or does gcovr (from the master branch) raise that AttributeError under Python 2.6?

   If so, what is `repr(options.gcov_exclude)` immediately before the patterns are compiled?

I have a hunch this isn't a Python 2.6 problem, but due to insufficient documentation of the `--gcov-exclude` and `--gcov-filter` options..

**Comment by nickbroon** [comment]
I'm seeing a similar traceback.

```gcovr -v --xml --keep --object-directory=obj-x86_64-linux-gnu -o coverage.xml```

```
04:15:05 [  416s] Traceback (most recent call last):
04:15:05 [  416s]   File "/usr/bin/gcovr", line 2312, in <module>
04:15:05 [  416s]     process_datafile(file_, covdata, options)
04:15:05 [  416s]   File "/usr/bin/gcovr", line 891, in process_datafile
04:15:05 [  416s]     process_gcov_data(fname, covdata, abs_filename, options)
04:15:05 [  416s]   File "/usr/bin/gcovr", line 500, in process_gcov_data
04:15:05 [  416s]     root_dir = os.path.abspath(options.root)
04:15:05 [  416s]   File "/usr/lib/python2.7/posixpath.py", line 360, in abspath
04:15:05 [  416s]     if not isabs(path):
04:15:05 [  416s]   File "/usr/lib/python2.7/posixpath.py", line 54, in isabs
04:15:05 [  416s]     return s.startswith('/')
04:15:05 [  416s] AttributeError: 'NoneType' object has no attribute 'startswith'
```

I'm using this version of gcovr: https://packages.debian.org/stretch/gcovr

**Comment by sinelaw** [comment]
Assuming this is really a python 2.6 issue, do we really want to fix it? It might make more sense to declare python 2.7 as a minimal requirement

**Comment by latk** [comment]
I don't think this is a Python 2.6 issue. The traceback into the Python standard library may be due to passing a regex into the `abspath()` function which expects a string, and would therefore be unrelated to the “no output” problem. I.e. the error is an artefact of the debugging approach.

The missing output may be due to a problem with the regexes used for the filters. Without the information requested in my [previous comment](https://github.com/gcovr/gcovr/issues/174#issuecomment-322773783) this can't be solved. I am therefore closing this issue for now, but will reopen when the necessary information is added.

How the filters are compiled was changed in #158, but I don't think that affects this issue.

@nickbroon Your issue occurs because you didn't specify an explicit `-r` root. This was resolved in commit e7d1bbe62181b003cf69517fb87994717a274184. The fix will ship with gcovr 3.4. Please consider testing the development version of gcovr to verify the fix.
